packages/beacon-node/src/execution/engine/http.ts

```diff
@@ -421,6 +435,40 @@ export class ExecutionEngineHttp implements IExecutionEngine {
     return this.state;
   }
 
+  async getClientVersion(clientVersion?: ClientVersion): Promise<ClientVersion[]> {
+    const method = "engine_getClientVersionV1";
+
+    const response = await this.rpc.fetchWithRetries<
+      EngineApiRpcReturnTypes[typeof method],
+      EngineApiRpcParamTypes[typeof method]
+    >({method, params: [clientVersion ?? this.getConsensusClientVersion()]});
```

Comment:
Need to confirm that we sent 8 bytes as the commit hash, I think we currently use 7 to match github hashes

Depending how strict the ELs validate this request it might cause issues, see related
- https://github.com/paradigmxyz/reth/pull/9137

Reply:
@nflaig I am confused with the discussion you linked.

From the [spec](https://github.com/ethereum/execution-apis/blob/main/src/engine/identification.md#clientversionv1), we should return only return 4 byte commit. 

In this PR, we get long commit from https://github.com/ChainSafe/lodestar/blob/e666c97e9c12b9d60036db1fd8ac6b5b35589ede/packages/cli/src/util/version.ts#L55 and later we slice it to 4 bytes here https://github.com/ChainSafe/lodestar/blob/e666c97e9c12b9d60036db1fd8ac6b5b35589ede/packages/beacon-node/src/util/graffiti.ts#L17

Reply:
is `commit` of type string here? if yes, then 4 bytes == 8 characters

Reply:
ok I think I am correct, sha1 hashes are hex-encoded strings

Reply:
Just want to clarify, if the CL/EL commit we send/receive through engine API are 4 bytes (8 characters).

The commits we use for constructing graffiti is 2 bytes (4 characters)

Reply:
Looks correct now